Yatima/ForLurkRepo/DSL.lean

```diff
@@ -177,4 +177,19 @@ end
 
 elab "⟦ " e:lurk_expr " ⟧" : term =>
   elabLurkExpr e
-  
\ No newline at end of file
+
+namespace Lurk.Expr 
+
+def mkMutualBlock (mutuals : List (Name × Expr)) : List (Name × Expr) :=
+  let names := mutuals.map Prod.fst
+  let mutualName := names.head! ++ `mutual
+  let fnProjs := names.enum.map fun (i, (n : Name)) => (n, app ⟦$mutualName⟧ [⟦$i⟧])
+  let targets := fnProjs.map fun (n, e) => (⟦$n⟧, e)
+  let mutualBlock := mkIfElses (mutuals.enum.map fun (i, n, e) => 
+    (⟦(= mutidx $i)⟧, e.replaceN targets)  
+  ) ⟦nil⟧
+  (mutualName, ⟦(lambda (mutidx) $mutualBlock)⟧) :: fnProjs
+
```

Comment:
This was my original attempts at creating mutuals before getting distracted by everything. It kinda works